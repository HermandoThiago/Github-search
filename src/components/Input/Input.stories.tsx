import { useState, ChangeEvent } from 'react';
import { Meta, Story } from '@storybook/react';
import Input, { IPropsInput } from './Input';

export default {
    title: 'Components/Input',
    component: Input,
    args: {
        placeholder: 'Name',
        type: 'text',
    }
} as Meta;

export const Default: Story<IPropsInput> = (args) => {
    return (
        <Input {...args} />
    );
}


