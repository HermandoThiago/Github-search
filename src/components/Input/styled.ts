import styled from "styled-components";

export const StyledInput = styled.input`
    width: 400px;
    background-color: #2A2550;
    border: 0;
    padding: 11px 20px;
    font-weight: lighter;
    border-radius: 6px;
    margin-right: 5px;

    :focus{
        outline: 0;
    }

    @media screen and (max-width: 600px){
        width: 100%;
    }

`;
