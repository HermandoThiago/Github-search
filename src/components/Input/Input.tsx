import { ChangeEvent } from "react";
import { StyledInput } from "./styled";

export interface IPropsInput {
  placeholder: string;
  type: string;
  change: (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
  value: string;
}

export function Input({
  placeholder = "placeholder",
  type = "text",
  change,
  value,
}: IPropsInput) {
  return (
    <StyledInput
      placeholder={placeholder}
      type={type}
      onChange={change}
      value={value}
    />
  );
}
