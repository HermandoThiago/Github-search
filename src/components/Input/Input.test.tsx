import { ChangeEvent } from 'react';
import renderer from 'react-test-renderer';
import Input from './Input';

describe('Tests Input', () => {
    it('should match snapshot', () => {
        let name = '';

        const handleName = (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
            name = e.target.value;
        };

        const component = renderer.create(
            <Input 
                placeholder='Name'
                type='text'
                value={name}
                change={handleName}
            />
        ).toJSON();

        expect(component).toMatchSnapshot();
    });
});
