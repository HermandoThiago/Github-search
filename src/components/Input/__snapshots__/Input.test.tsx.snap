// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Input should match snapshot 1`] = `
<input
  className="sc-bczRLJ dslVns"
  onChange={[Function]}
  placeholder="Name"
  type="text"
  value=""
/>
`;
